client/src/components/QAhelperFunctions.js

```diff
@@ -0,0 +1,34 @@
+/* eslint-disable no-var */
+const _ = require('underscore');
+//combines the text of the Q and its A's together, then filters them based on the search bar entry
+
+const reviewFilter = function (Qs, filter) {
+  //An object with the the Q ids for keys, and a single string of the Q/A text as a value
+  const combinedQAObj = {};
+  //iterate through each q for the given Product
+  Qs.results.forEach((q) => {
+    //get the question's id
+    const id = q.question_id;
+    //Then iterates an array of all the A's texts for that Q,
+    var answers = Object.values(q.answers).map((val) => (
+    //Map to return an array of the answer texts
+      (answers === undefined ? val.body : answers + val.body)
+    ));
+      //Store joined Q/A texts string aside Q_id in the Obj
+    combinedQAObj[id] = q.question_body + answers.join('');
+  });
+  //use the combined strings for each question to filter out the qs by search term
+  return Qs.results.filter((q) => (combinedQAObj[q.question_id].indexOf(filter) !== -1));
+};
+
```

Comment:
this function calls the previous function IF there is a filter warranted (over two characters in the search bar)
If it is not called then it puts the question object into the same format that it would have recieved from the other function
It then sorts by helpfulness rating